package/base-files/files/etc/uci-defaults/14_migrate-swconfig

```diff
@@ -0,0 +1,72 @@
+findPorts()
+# Differentiate ports between user and cpu ports. Display DSA interface name of user ports.
+# Check compiled devicetree to do this.
+
+{
+	local port=$1
+
+	# Devicetree path is currently hardcoded for Linksys WRT32X.
+	# It should be changed to automatically find the "ports/port@" directory in "/proc/device-tree".
+	# Check dsa-port.yaml for other accepted node names like "ethernet-ports/ethernet-port@".
+
+	# A port may be missing from DSA bindings (it's usually the second CPU port), ignore these ports.
+	if [[ ! -d "/proc/device-tree/soc/internal-regs/mdio@72004/switch@0/ports/port@${port}" ]]
```

Comment:
`mdio@72004` tells me this is targeted for specific hardware

so what hardware are you testing with?

you need to either 

A. move this to the base-files include folder of a specific target
or
B. add autodetection of the mdio path for each target this would be used for

Reply:
> `mdio@72004` tells me this is targeted for specific hardware

This is WIP, it was hardcoded to 
see if the rest of the script worked. I believe I already explain this in the comments.

> 
> 
> 
> so what hardware are you testing with?

Please read the commented lines.

> 
> 
> 
> you need to either 
> 
> 
> 
> A. move this to the base-files include folder of a specific target
> 
> or
> 
> B. add autodetection of the mdio path for each target this would be used for
> 
> 
> 
> (edit)
> 
> 
> 
> > * Find devicetree path automatically.
> 
> 
> 
> yeah this

There's a lot of things to work on. Anyone is welcome to do this part.

Reply:
sorry, when I say "hardware" i mean the target, which is not in comments

after some looking, the target is mvebu/cortexa9 correct or no?

that's where this script should be. I think its best to handle this on a per target basis, because you already will have variation between boards within that target